src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/UrlSegment.cs

```diff
@@ -1,13 +1,15 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.AspNetCore.Rewrite.Extensions;
+
 namespace Microsoft.AspNetCore.Rewrite.Internal.PatternSegments
 {
     public class UrlSegment : PatternSegment
     {
         public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
         {
-            return context.HttpContext.Request.Path;
+            return context.GlobalRule ? context.HttpContext.Request.ToUri().AbsoluteUri : (string)context.HttpContext.Request.Path;
```

Comment:
i agree with you and that's what i started to do. `IISUrlRewriteRule` does have the property set but the issue i ran into is that inside of `UrlSegment`, doesn't have visibility into the rule. i thought about pushing the property in explicitly but that would've made a sweeping mess. i figured it was easiest/best given the current design to piggyback on the context as it required minimal changes. happy to discuss other ideas.